### DISCUSSION TITLE
Conveyor Belts

### DISCUSSION THREAD

**Discussion by PhoenixIllusion** [jrouwe/JoltPhysics.js#38] [Show and tell]
I tried to do a 1-to-1 port of the conveyor belt test from JoltPhysics.
Original: https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/ConveyorBeltTest.cpp

Port: https://github.com/PhoenixIllusion/JoltPhysics.js/blob/feature/conveyor_belt/Examples/conveyor_belt.html

![2023-10-14-conveyor-belt-01](https://github.com/jrouwe/JoltPhysics.js/assets/756488/f1b77898-d42f-493a-8bdd-c442402a7756)

I'm a little confused on the positioning values of the CPP project.
I had to modify the Y axis of those initial gray blocks and the entire heights of the 4x belts and everything to get it like this.
I do not have a Windows machine so wasn't able to run the initial test, but I'm not sure why all the boxes are marked as having a Z position at the far side of the room, off most of the belts.
`cargo_settings.mPosition = RVec3(-cBeltLength + i * 10.0f, 10.0f, -cBeltLength);`

I'm not sure if the blue belt is supposed to crawl its way off the screen and fall off the floor?

The Test code seems to have all the old boxes lined up on the far side, and the red angular belt is 2-3 units right under the center so nothing lands on it?

![2023-10-14-conveyor-belt-01-old](https://github.com/jrouwe/JoltPhysics.js/assets/756488/1b4116a9-b1e2-46e8-914e-ed95be9f8e14)

**Comment by jrouwe** [comment]
The demo looks a bit different, this is what it is supposed to look like:

https://www.youtube.com/watch?v=p_H6egZzbZE&list=PLYXVwtOr1CBxbA50jVg2dKUQvHW_5OOom&index=23

There were a couple of small mistakes that I fixed and now it looks like the Jolt demo. I merged the demo into master. Again, thanks for the demo!
